Sources/SwiftDocC/Model/MarkdownOutput/Translation/MarkdownOutputSemanticVisitor.swift

```diff
@@ -0,0 +1,442 @@
+            _ = visit(media)
+        }
+        return nil
+    }
+    
+    mutating func visitIntro(_ intro: Intro) -> MarkdownOutputNode? {
+        
+        markdownWalker.visit(Heading(level: 1, Text(intro.title)))
+        
+        for child in intro.children {
+            _ = visit(child)
+        }
+        return nil
+    }
+    
+    mutating func visitMarkupContainer(_ markupContainer: MarkupContainer) -> MarkdownOutputNode? {
+        markdownWalker.withRemoveIndentation(from: markupContainer.elements.first) {
+            $0.visit(container: markupContainer)
+        }
+        return nil
+    }
+    
+    mutating func visitImageMedia(_ imageMedia: ImageMedia) -> MarkdownOutputNode? {
+        markdownWalker.visit(imageMedia)
+        return nil
+    }
+    
+    mutating func visitVideoMedia(_ videoMedia: VideoMedia) -> MarkdownOutputNode? {
+        markdownWalker.visit(videoMedia)
+        return nil
+    }
+    
+    mutating func visitContentAndMedia(_ contentAndMedia: ContentAndMedia) -> MarkdownOutputNode? {
+        for child in contentAndMedia.children {
+            _ = visit(child)
+        }
+        return nil
+    }
+    
+    mutating func visitCode(_ code: Code) -> MarkdownOutputNode? {
+        // Code rendering is handled in visitStep(_:)
+        return nil
+    }
+}
+
+
+// MARK: Visitors not currently used for markdown output
+extension MarkdownOutputSemanticVisitor {
+        
+    mutating func visitXcodeRequirement(_ xcodeRequirement: XcodeRequirement) -> MarkdownOutputNode? {
+        return nil
+    }
+    
+    mutating func visitAssessments(_ assessments: Assessments) -> MarkdownOutputNode? {
+        return nil
+    }
+    
+    mutating func visitMultipleChoice(_ multipleChoice: MultipleChoice) -> MarkdownOutputNode? {
+        return nil
+    }
+    
+    mutating func visitJustification(_ justification: Justification) -> MarkdownOutputNode? {
+        return nil
+    }
+    
+    mutating func visitChoice(_ choice: Choice) -> MarkdownOutputNode? {
+        return nil
+    }
+        
+    mutating func visitTechnology(_ technology: TutorialTableOfContents) -> MarkdownOutputNode? {
+        return nil
+    }
+        
+    mutating func visitVolume(_ volume: Volume) -> MarkdownOutputNode? {
+        return nil
+    }
+    
+    mutating func visitChapter(_ chapter: Chapter) -> MarkdownOutputNode? {
+        return nil
+    }
+    
+    mutating func visitTutorialReference(_ tutorialReference: TutorialReference) -> MarkdownOutputNode? {
+        return nil
+    }
+    
+    mutating func visitResources(_ resources: Resources) -> MarkdownOutputNode? {
+        return nil
+    }
+    
+    mutating func visitTile(_ tile: Tile) -> MarkdownOutputNode? {
+        return nil
+    }
+    
+    mutating func visitComment(_ comment: Comment) -> MarkdownOutputNode? {
+        return nil
+    }
+    
+    mutating func visitTutorialArticle(_ article: TutorialArticle) -> MarkdownOutputNode? {
```

Comment:
Tutorial articles are full of human written documentation content. Is there a reason for excluding them completely from the output?